speech/google/cloud/speech/client.py

```diff
@@ -261,3 +377,160 @@ def _build_request_data(sample, language_code=None, max_alternatives=None,
+                           is more likely to recognize them. This can be
+                           used to improve the accuracy for specific words
+                           and phrases. This can also be used to add new
+                           words to the vocabulary of the recognizer.
+
+    :type single_utterance: boolean
+    :param single_utterance: [Optional] If false or omitted, the recognizer
+                             will perform continuous recognition
+                             (continuing to process audio even if the user
+                             pauses speaking) until the client closes the
+                             output stream (gRPC API) or when the maximum
+                             time limit has been reached. Multiple
+                             SpeechRecognitionResults with the is_final
+                             flag set to true may be returned.
+
+                             If true, the recognizer will detect a single
+                             spoken utterance. When it detects that the
+                             user has paused or stopped speaking, it will
+                             return an END_OF_UTTERANCE event and cease
+                             recognition. It will return no more than one
+                             SpeechRecognitionResult with the is_final flag
+                             set to true.
+
+    :type interim_results: boolean
+    :param interim_results: [Optional] If true, interim results (tentative
+                            hypotheses) may be returned as they become
+                            available (these interim results are indicated
+                            with the is_final=false flag). If false or
+                            omitted, only is_final=true result(s) are
+                            returned.
+    """
+    config_request = _make_streaming_config(
+        sample, language_code=language_code, max_alternatives=max_alternatives,
+        profanity_filter=profanity_filter, speech_context=speech_context,
+        single_utterance=single_utterance, interim_results=interim_results)
+
+    # The config request MUST go first and not contain any audio data.
+    yield config_request
+
+    while True:
+        data = sample.stream.read(sample.chunk_size)
+        if not data:
+            break
+        # Optimize the request data size to around 100ms.
+        yield StreamingRecognizeRequest(audio_content=data)
+
+
+def _make_streaming_config(sample, language_code,
+                           max_alternatives, profanity_filter,
+                           speech_context, single_utterance,
+                           interim_results):
+    """Build streaming configuration.
+
+    :type sample: :class:`~google.cloud.speech.sample.Sample`
+    :param sample: Instance of ``Sample`` containing audio information.
+
+    :type language_code: str
+    :param language_code: (Optional) The language of the supplied audio as
```

Comment:
I realize this was copy-pasted but **none** of the arguments to this function are optional.